libminifi/include/utils/ThreadPool.h

```diff
@@ -330,20 +331,18 @@ class ThreadPool {
 // integrated power manager
   std::shared_ptr<controllers::ThreadManagementService> thread_manager_;
   // thread queue for the recently deceased threads.
-  moodycamel::ConcurrentQueue<std::shared_ptr<WorkerThread>> deceased_thread_queue_;
+  ConcurrentQueue<std::shared_ptr<WorkerThread>> deceased_thread_queue_;
 // worker queue of worker objects
-  moodycamel::ConcurrentQueue<Worker<T>> worker_queue_;
+  ConditionConcurrentQueue<Worker<T>> worker_queue_;
   std::priority_queue<Worker<T>, std::vector<Worker<T>>, DelayedTaskComparator<T>> delayed_worker_queue_;
-// notification for available work
-  std::condition_variable tasks_available_;
+// mutex to  protect task status and delayed queue   
+  std::mutex worker_queue_mutex_;
```

Comment:
Is this mutex still used?

Reply:
Yes, still used to protect the delayed queue and the status of current tasks.

Reply:
Removed